server/devpi_server/views.py

```diff
@@ -1446,6 +1446,104 @@ def user_list(self):
             d[user.name] = user.get()
         apireturn(200, type="list:userconfig", result=d)
 
+    @view_config(
+        route_name="/+snapshot/{user}/{index}", request_method="POST")
+    @view_config(
+        route_name="/+snapshot/{user}/{index}/", request_method="POST")
+    def snapshot(self):
+        request = self.request
+        context = self.context
+        if not request.has_permission("upload"):  # TODO: add 'snapshot' perm ?
```

Comment:
I would use ``index_create`` for now.

Separate permissions might make sense, but then there should be ``index_copy`` enabled by default and ``index_snapshot`` disabled by default, as a admin might not want hard links to be used.

Reply:
effectively `index_create` seems even more legit than upload. 

for an admin not wanting to use hard links by default.. I'm not sure to understand.. why would he prefer to have duplicate/copy of the wheels/release files instead of doing hard-links ?